File: my-folder/1028-interval-list-intersections/solution.java
class Solution {
    public int[][] intervalIntersection(int[][] A, int[][] B) {
        List<int[]> ans = new ArrayList<>();
        int i = 0, j = 0;
        while (i < A.length && j < B.length) {
            int low = Math.max(A[i][0], B[j][0]);
            int high = Math.min(A[i][1], B[j][1]);
            if (low <= high) {
                ans.add(new int[] {low, high});
            }
            
            if (A[i][1] < B[j][1]) {
                i++;
            } else {
                j++;
            }
        }
        
        return ans.toArray(new int[ans.size()][]);
    }
}

File: my-folder/1028-interval-list-intersections/solution.py
class Solution:
    def intervalIntersection(self, firstList: List[List[int]], secondList: List[List[int]]) -> List[List[int]]:
        res = []
        i,j = 0, 0
        while i < len(firstList) and j < len(secondList):
            a1,a2 = firstList[i][0], firstList[i][1]
            b1,b2 = secondList[j][0], secondList[j][1]
            
            if b2 >= a1 and a2 >= b1:
                res.append([max(a1,b1), min(a2,b2)])
            if b2 < a2:
                j += 1
            else:
                i += 1
        return res
